src/client/driver/command-executors/execute-child-window-selector.js

```diff
@@ -0,0 +1,21 @@
+import hammerhead from '../deps/hammerhead';
+import { ChildWindowNotFoundError } from '../../../errors/test-run';
+
+const Promise = hammerhead.Promise;
+
+export default function executeChildWindowSelector (selector, childWindowLinks) {
+    if (typeof selector === 'string') {
+        const foundChildWindowDriverLink = childWindowLinks.filter(link => link.pageId === selector)[0];
+
+        if (!foundChildWindowDriverLink) {
+            const error = new ChildWindowNotFoundError();
+
+            throw error;
```

Comment:
Should this error be caught in the outer catch block? Now it will not be caught.

Reply:
Yes, we will catch the error in another code after the `t.switchToWindow` action is implemented.